From khali@linux-fr.org Mon Jun 12 12:51:28 2006
Date: Mon, 12 Jun 2006 21:51:20 +0200
From: Jean Delvare <khali@linux-fr.org>
To: Greg KH <greg@kroah.com>
Cc: LM Sensors <lm-sensors@lm-sensors.org>
Subject: [PATCH 08/14] i2c-i801: Remove PCI function check
Message-Id: <20060612215120.0578fd85.khali@linux-fr.org>
Content-Disposition: inline; filename=i2c-i801-03-remove-pci-function-check.patch

i2c-i801: Remove PCI function check

Remove the PCI function number check when probing devices.
This check is redundant, each function has a separate PCI device
ID, so checking for that ID is sufficient.

Signed-off-by: Jean Delvare <khali@linux-fr.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/i2c/busses/i2c-i801.c |    4 ----
 1 file changed, 4 deletions(-)

--- gregkh-2.6.orig/drivers/i2c/busses/i2c-i801.c
+++ gregkh-2.6/drivers/i2c/busses/i2c-i801.c
@@ -107,10 +107,6 @@ static int i801_setup(struct pci_dev *de
 	int error_return = 0;
 	unsigned char temp;
 
-	/* Note: we keep on searching until we have found 'function 3' */
-	if(PCI_FUNC(dev->devfn) != 3)
-		return -ENODEV;
-
 	I801_dev = dev;
 	if ((dev->device == PCI_DEVICE_ID_INTEL_82801DB_3) ||
 	    (dev->device == PCI_DEVICE_ID_INTEL_82801EB_3) ||
